aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -144,6 +144,84 @@ Tensor& binary_cross_entropy_out_cuda(const Tensor& input, const Tensor& target,
   return loss;
 }
 
+namespace {
+Tensor& binary_cross_entropy_with_logits_out_cuda(
+    const Tensor& input,
+    const Tensor& target,
+    const c10::optional<Tensor>& weight_opt,
+    const c10::optional<Tensor>& pos_weight_opt,
+    int64_t reduction,
+    Tensor& loss) {
+// See [Note: hacky wrapper removal for optional tensor]
+  c10::MaybeOwned<Tensor> weight_maybe_owned = at::borrow_from_optional_tensor(weight_opt);
+  const Tensor& weight = *weight_maybe_owned;
+  c10::MaybeOwned<Tensor> pos_weight_maybe_owned = at::borrow_from_optional_tensor(pos_weight_opt);
+  const Tensor& pos_weight = *pos_weight_maybe_owned;
+
+  Tensor loss_squeezed = at::squeeze(loss);
+  if (pos_weight_opt.has_value()) {
+    auto iter = TensorIteratorConfig()
+      .add_output(loss_squeezed)
+      .add_owned_input(at::squeeze(input))
+      .add_owned_input(at::squeeze(target))
+      .add_owned_input(at::squeeze(pos_weight))
+      .build();
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, loss.scalar_type(), "binary_cross_entropy_with_logits_out_cuda", [&] {
+      gpu_kernel(
+          iter,
+          [] GPU_LAMBDA (scalar_t input_val, scalar_t target_val, scalar_t pos_weight_val) -> scalar_t {
+            auto max_val = std::max(-input_val, static_cast<scalar_t>(0));
+            auto log_weight = (pos_weight_val - 1) * target_val + 1;
+            auto weight_multiplier = std::log(std::exp(-max_val) + std::exp(-input_val - max_val)) + max_val;
+            return (1 - target_val) * input_val + (log_weight * weight_multiplier);
+          }
+      );
+    });
+  } else {
+    auto iter = TensorIteratorConfig()
+      .add_output(loss_squeezed)
+      .add_owned_input(at::squeeze(input))
+      .add_owned_input(at::squeeze(target))
+      .build();
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, loss.scalar_type(), "binary_cross_entropy_with_logits", [&] {
+      gpu_kernel(
+          iter,
+          [] GPU_LAMBDA (scalar_t input_val, scalar_t target_val) -> scalar_t {
+            auto max_val = std::max(-input_val, static_cast<scalar_t>(0));
+            return (1 - target_val) * input_val + max_val + std::log(std::exp(-max_val) + std::exp(-input_val - max_val));
+          }
+      );
+    });
+  }
+  if (weight.defined()) {
+    loss.mul_(weight);
+  }
+
+  if (reduction != at::Reduction::None) {
+    Tensor loss_reduced;
+    if (reduction == at::Reduction::Mean) {
+      loss_reduced = loss.mean();
+    } else if (reduction == at::Reduction::Sum) {
+      loss_reduced = loss.sum();
+    }
+    loss.resize_as_(loss_reduced).copy_(loss_reduced);
```

Comment:
looks like you should be able to avoid this copy? You are not obligated to have `*_out_cuda` function that has to get real `loss` argument that you can modify only inplace, so you can just return `loss_reduced`.